modules/tinymce/src/core/main/ts/content/InsertContentImpl.ts

```diff
@@ -22,6 +22,19 @@ import * as SelectionUtils from '../selection/SelectionUtils';
 import { InsertContentDetails } from './ContentTypes';
 import * as InsertList from './InsertList';
 
+const wrappedElements = [ 'pre' ];
+
+const shouldFlatten = (fragment: AstNode, parentNode: Element) => {
+  const currentNode = fragment.firstChild;
+
+  const isAFlattenableTag = Arr.contains(wrappedElements, currentNode.name);
+  const isPastingInTheSameTag = currentNode.name === parentNode.tagName.toLowerCase();
+  // The last children is the bookmarkHtml attached by the function insertHtmlAtCaret
+  const isCopingOnlyOneTag = currentNode === fragment.lastChild.prev;
```

Comment:
Should this double check that the lastChild is the bookmark span? As this could still lead to unexpected cases e.g:
```suggestion
const lastNode = fragment.lastChild.attr('data-mce-type') === 'bookmark' ? fragment.lastChild.prev : fragment.lastChild;
const isCopingOnlyOneTag =  currentNode === lastNode;
```

Reply:
done